napari/components/viewer_model.py

```diff
@@ -282,22 +279,29 @@ def interactive(self, interactive):
     def active_layer(self):
         """int: index of active_layer
         """
-        return self._active_layer
+        warnings.warn(
+            (
+                "The viewer.active_layer property is deprecated and will be removed after version 0.4.5."
+                " Instead you should use the viewer.layers.active property"
+            ),
+            category=DeprecationWarning,
+            stacklevel=2,
+        )
+        return self.layers.active
```

Comment:
Why its return `self.layers.active` if there is an idea for a cache for `qt_viewer`. It should use the same cache. 

```suggestion
        warnings.warn(
            (
                "The viewer.active_layer property is deprecated and will be removed after version 0.4.5."
                " Instead you should use the viewer.layers.active property"
            ),
            category=DeprecationWarning,
            stacklevel=2,
        )
        return self._active_layer
```